nemo/lightning/_strategy_lib.py

```diff
@@ -514,6 +514,19 @@ def get_safe(param_id):
 
 def load_model_state_dict(megatron_parallel, checkpoint: Mapping[str, Any], strict: bool = True) -> None:
     from megatron.core import parallel_state
+    from megatron.core.dist_checkpointing.validation import StrictHandling, parse_strict_flag
+
+    strict = parse_strict_flag(strict)
+    ## convert from StrictHandling to bool for PTL
+    if strict is not None and not isinstance(strict, bool):
+        strict_options = [
+            StrictHandling.RAISE_UNEXPECTED,
+            StrictHandling.RAISE_ALL,
```

Comment:
I think we should also add `StrictHandling.ASSUME_OK_UNEXPECTED`?
Since this is the default one and we don't expect mismatches